### PR TITLE
Add other nuts samplers example notebook

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/pymc-labs/pymc-marketing/issues/577

This should help users use other samplers which generally speed inference.

----
📚 Documentation preview 📚: https://pymc-marketing--579.org.readthedocs.build/en/579/

### CODE REVIEW

File: docs/source/notebooks/general/other_nuts_samplers.ipynb [hunk 1]
```diff
@@ -0,0 +1,336 @@
+{
```
Comment (twiecki) [inline docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>Maybe add that these samplers can be significantly faster and also sample on the GPU.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/579/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (twiecki) [inline docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>Add nutpie example?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/579/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (juanitorduz) [reply docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: Added!

Comment (juanitorduz) [reply docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>Sure! Added!</p>

File: docs/source/notebooks/index.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
-(clv_examples)=
-# Examples
+(howto)=
+# How to?
```
Comment (wd60622) [inline docs/source/notebooks/index.md:2]: Thoughts on without How to without question mark?
ie.
- How to 
- How-to

File: docs/source/notebooks/general/other_nuts_samplers.ipynb [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
```
Comment (wd60622) [inline docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>Would it be helpful to have a hyperlink to docs? </p><p><a href="https://www.pymc.io/projects/docs/en/stable/api/generated/pymc.sampling.mcmc.sample.html#pymc.sampling.mcmc.sample" target="_blank">https://www.pymc.io/projects/docs/en/stable/api/generated/pymc.sampling.mcmc.sample.html#pymc.sampling.mcmc.sample</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/579/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (wd60622) [inline docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>maybe pull out the constant kwargs in order to emphasize the nuts_sampler and minimize moving parts. </p><p>i.e. </p><p><code></code>`python</p><p>kwargs = {</p><p>    "draws": 2000, </p><p>    "chains": 5, </p><p>    "target_accept": 0.9, </p><p>    "random_seed": 42, </p><p>}</p><p> </p><p>idata_numpyro = model.fit(nuts_sampler="numpyro", **kwargs)</p><p>idata_blackjax = model.fit(nuts_sampler="blackjax", **kwargs)</p><p>...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/579/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: docs/source/notebooks/index.md (hunk 2 above)
Comment (juanitorduz) [reply docs/source/notebooks/index.md:2]: +1

File: docs/source/notebooks/general/other_nuts_samplers.ipynb (hunk 3 above)
Comment (juanitorduz) [reply docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>yes!</p>

Comment (wd60622) [reply docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: "its so easy to do 😄 "

Comment (juanitorduz) [reply docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>Indeed!</p>

Comment (juanitorduz) [reply docs/source/notebooks/general/other_nuts_samplers.ipynb:1]: <p>Yes!</p>

**Review by wd60622 — APPROVED** [review]
Looks great! Only think I might add would be to actually time the sampling. Just to prove to the users that it's faster. I don't see a progressbar in the rendered docs

**Comment by wd60622** [comment]
Side question: How do you get the collapsible section from the notebook? I don't see anything signifying that in the notebook...
Screenshot 2024-03-09 at 20 41 11

**Comment by juanitorduz** [comment]
> Looks great! Only think I might add would be to actually time the sampling. Just to prove to the users that it's faster. I don't see a progressbar in the rendered docs

I just wanted to wide some numba warnings and messages which might distract the target audience. Just to keep it simple.

**Comment by juanitorduz** [comment]
> Side question: How do you get the collapsible section from the notebook? I don't see anything signifying that in the notebook... Screenshot 2024-03-09 at 20 41 11

If you are using vscode check the cell options and click "add tag" and simply write "hide-input" in the pop-out window , see https://myst-nb.readthedocs.io/en/latest/render/hiding.html :)

**Comment by wd60622** [comment]
> I just wanted to wide some numba warnings and messages which might distract the target audience. Just to keep it simple.

Sounds good. Lgtm!

**Review by twiecki — APPROVED** [review]

**Pull request merged**